metricflow/dataflow/optimizer/source_scan/cm_branch_combiner.py

```diff
@@ -293,6 +293,14 @@ def visit_compute_metrics_node(self, node: ComputeMetricsNode) -> ComputeMetrics
             )
             return ComputeMetricsBranchCombinerResult()
 
+        if self._current_left_node.for_group_by_source_node != current_right_node.for_group_by_source_node:
```

Comment:
Can you move this check (and the above `aggregated_to_elements` check) to a method similar to:

```
def combine(other: ComputeMetricsNode) -> CombinationResult
    ...
```

That should make it easier update that node in the future.

Reply:
Might be easier to do `can_combine` instead for now.